mono/mini/mini-wasm-debugger.c

```diff
@@ -662,6 +681,49 @@ static gboolean describe_value(MonoType * type, gpointer addr)
 			}
 			break;
 		}
+		case MONO_TYPE_VALUETYPE:
+		case MONO_TYPE_TYPEDBYREF: {
+			MonoClass *klass = mono_class_from_mono_type_internal (type);
+			if (klass == datetime_class) {
+				MonoObject *obj = *(MonoObject**)addr;
+				if (!obj) {
+					mono_wasm_add_string_var (NULL);
+				} else {
+					int objectid = get_object_id(obj);
+					mono_wasm_add_obj_var ("System.DateTime", objectid);
+					g_hash_table_insert_replace (obj_to_val_refs, GINT_TO_POINTER(objectid), addr, TRUE);
+					g_hash_table_insert_replace (val_to_types, GINT_TO_POINTER(objectid), GINT_TO_POINTER(1), TRUE);
+				}
+			} else if (klass == datetimeoffset_class) {
+				MonoObject *obj = *(MonoObject**)addr;
+				if (!obj) {
+					mono_wasm_add_string_var (NULL);
+				} else {
+					int objectid = get_object_id(obj);
+					mono_wasm_add_obj_var ("System.DateTimeOffset", objectid);
+					g_hash_table_insert_replace (obj_to_val_refs, GINT_TO_POINTER(objectid), addr, TRUE);
+					g_hash_table_insert_replace (val_to_types, GINT_TO_POINTER(objectid), GINT_TO_POINTER(1), TRUE);
+				}
+			} else if (klass == guid_class) {
+				MonoObject *obj = *(MonoObject**)addr;
+				if (!obj) {
+					mono_wasm_add_string_var (NULL);
+				} else {
+					int objectid = get_object_id(obj);
+					mono_wasm_add_obj_var ("System.Guid", objectid);
+					g_hash_table_insert_replace (obj_to_val_refs, GINT_TO_POINTER(objectid), addr, TRUE);
```

Comment:
addr is a temporary value which is only valid as long as the debuggee is suspended. For this reason, it shouldn't be stored in hash tables etc. Instead, the whole valuetype is usually sent to the IDE in the normal non-wasm debugger.

Reply:
This has proven to be a problem so far.  Saw this in the debugger agent but so far the interaction messages are different.  Will look into this more.